File: lab6/lab6.py
def init():
    number = int(input("Enter the number to check every perfect number before it: "))
    return number

def divisors_sum(num):
    sum = 0
    for i in range(1, num): 
        if num % i == 0:
            sum += i
    return sum

def output():
    for k in range(1, number+1):
        sumk = divisors_sum(k)
        if sumk == k:
            print(k)
number = init()
output()

File: lab6/lab6.cpp
#include <iostream>
using namespace std;
void Input();
int Divisors_sum(int);
void Output();
int number;

int main()
{
    Input();
    Output();
}

void Input()
{
    cout << "Enter the number to count every perfect number before it: ";
    cin >> number;
}

int Divisors_sum(int n)
{
    int sum = 0;
    for(int i = 1; i < n; i++)
    {
        if(n % i == 0)
        {
            sum += i;
        }
    }
    return sum;
}

void Output()
{
    for(int j = 1; j <= number; j++)
    {
        if(j == Divisors_sum(j))
        {
            cout << j << endl;
        }
    }
}
